src/main/java/com/github/manolo8/darkbot/Main.java

```diff
@@ -144,6 +144,19 @@ public Main(StartupParams params) {
         API.createWindow();
         if (params.getAutoStart()) setRunning(true);
         start();
+        if (params.getAutoHide()) {
+            Thread hide = new Thread(() -> {
```

Comment:
Implement this in the API adapter not in main, consider that it may be better to simply modify darkboat api to handle this rather than having a hacky 2nd thread that will keep on waiting until it's valid to set it to minimized, but regardless of that you can likely have the isValid method in DarkboatApiAdapter make the check and after returning true sending a setMinimized and modifying an internal state.

Keep in mind that `setMinimized` does NOT work on all systems and there's extra logic required to use either setMinimized or setVisible depending on a user setting for fully hide API, that this must respect